src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -13105,6 +13196,42 @@ public void ReturnThis_02(LanguageVersion languageVersion)
                 Diagnostic(ErrorCode.ERR_RefReturnStructThis, "this").WithLocation(5, 32));
         }
 
+        [Fact, WorkItem(63526, "https://github.com/dotnet/roslyn/issues/63526")]
+        public void ReturnThis_03()
+        {
+            var source = """
+                using System.Diagnostics.CodeAnalysis;
+
+                ref struct S2 {
+                    public S S;
+                }
+
+                ref struct S {
+                    public int field;
+                    public ref int refField;
+
+                    ref int Prop1 => ref field; // 1
+
+                    [UnscopedRef] ref int Prop2 => ref field; // okay
+
+                    S2 Prop3 => new S2 { S = this }; // Okay
+
+                    S Prop4 => new S { refField = ref this.field }; // 2
+
+                    [UnscopedRef] S Prop5 => new S { refField = ref this.field }; // okay
+                }
+                """;
+
+            var comp = CreateCompilation(new[] { source, UnscopedRefAttributeDefinition }, runtimeFeature: RuntimeFlag.ByRefFields);
+            comp.VerifyDiagnostics(
+                // (11,26): error CS8170: Struct members cannot return 'this' or other instance members by reference
+                //     ref int Prop1 => ref field; // 1
+                Diagnostic(ErrorCode.ERR_RefReturnStructThis, "field").WithLocation(11, 26),
+                // (17,24): error CS8170: Struct members cannot return 'this' or other instance members by reference
+                //     S Prop4 => new S { refField = ref this.field }; // 2
+                Diagnostic(ErrorCode.ERR_RefReturnStructThis, "refField = ref this.field").WithLocation(17, 24));
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider adding corresponding tests for instance methods that return ref structs with ref fields:
```csharp
S F4() => new S { refField = ref this.field }; // 2
[UnscopedRef] S F5() => new S { refField = ref this.field }; // okay
```

And consider adding tests for static methods:
```csharp
static S F4(S r) => new S { refField = ref r.field }; // 2
static S F5([UnscopedRef] S r) => new S { refField = ref r.field }; // okay
```

Reply:
I think you meant to have `ref S r`, in the static method case since that's needed to be able to return `ref r.field`.

I'm going to skip the intermediate local case, I feel we've reliably covered that ref locals get their RSTE from their initializer.